ocaml/xapi/xapi_pbd.ml

```diff
@@ -122,7 +122,10 @@ let resync_cluster_stack_for_sr_type ~__context ~sr_sm_type ~host =
   (* check cluster_host associated with both the host and a cluster with a matching cluster_stack *)
     let self  = Db.Cluster_host.get_cluster ~__context ~self:cluster_host in
     if List.mem (Db.Cluster.get_cluster_stack ~__context ~self) required_cluster_stacks
-    then Xapi_cluster_host.resync_host ~__context ~host
+    then begin
+      Xapi_cluster_host.create_as_necessary ~__context ~host;
```

Comment:
I don't think that the object can be missing here, it got created during the startup sequence, or during join, this one probably doesn't need to be changed, and calling just resync_host is fine.